agent/consul/state/acl.go

```diff
@@ -894,6 +900,12 @@ func (s *Store) aclTokenSetTxn(tx *memdb.Txn, idx uint64, token *structs.ACLToke
 		}
 	}
 
+	if prohibitUnprivileged {
+		if numValidRoles == 0 && numValidPolicies == 0 && len(token.ServiceIdentities) == 0 {
```

Comment:
### Background (my original thoughts) ###

Not that we like `acl.default_policy = "allow"` but wont prohibiting tokens that link to nothing break some things around that default policy.

Particularly if the default is allow, the anon token is configured to deny everything and then you create some tokens without any roles or policies. Currently, in that scenario the existence of a valid token grants implicit access to everything without having to configure other rules. With this change creating those types of tokens would not be possible right?

### Current Thinking ###
I think I might be reading this wrong actually. Since it would default to false and you haven't modified the HTTP API then if you were creating those tokens they would continue to work. You just wouldn't be able to do the same sort of thing with the login process (which seems reasonable).

Reply:
This only applies when the `ProhibitUnprivileged` setting is passed in, which is only going to come in on Login, which prevents tokens from being created that will have no roles links or service identities.